libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -574,7 +604,8 @@ AP_GPS::update_instance(uint8_t instance)
         return;
     }
 
-    if (drivers[instance] == nullptr || state[instance].status == NO_GPS) {
+    // If it is UAVCAN GNSS - give it regular time as if it is there to acknowledge itself
+    if (drivers[instance] == nullptr || (state[instance].status == NO_GPS && _type[instance] != GPS_TYPE_UAVCAN)) {
```

Comment:
As I've said in the other comment, the changes there are breaking the current code logic. Adding this check just for one driver is a hack and should be removed.